### DISCUSSION TITLE
Ability to modify and delete documents

### DISCUSSION THREAD

**Discussion by BorisFromBE** [openkfw/TruBudget#1144] [Ideas]
Currently, TruBudget allows users to upload one or several documents to a workflow item. However, once the document is uploaded and the workflow item submitted, the user cannot delete the document. It could become obsolete or its confidentiality could be reassessed. The only option is to upload another document which bloats the platform, might confuse the user when validating the workflow item based on document(s) and does not solve the confidentiality problem. 

We would like to allow a user to delete a document from an existing workflow item (see mockup).

![image](https://user-images.githubusercontent.com/91593570/168256691-12105c2f-d8ba-4333-9b2a-e919dcc66db9.png)

**Comment by Stezido** [comment]
A complete delete operation in a blockchain-based environment is not possible.
The way we could implement the delete function tough is to mark a document as deleted/removed on chain.
This way the blockchain still keeps track of the document addition and deletion and the frontend can only show the "not deleted"-documents. 
The frontend is a visualisation of Trubudget's api and should not hide information so I suggest to still make it possible to show deleted documents. I suggest to put the documents into two collapsible columns - categories like "attached" and "deleted" documents.
An example showing this collapsible behaviour what I mean is shown here: https://mui.com/material-ui/react-accordion/#customization
